tests/test_pcap.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License.  You may obtain a copy of
+# the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations under
+# the License.
+# ==============================================================================
+"""
+Test PcapDataset
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import tensorflow as tf
+
+tf.compat.v1.disable_eager_execution()
+
+import tensorflow_io.pcap as pcap_io # pylint: disable=wrong-import-position
+
+def test_pcap_input():
+    """test_pcap_input
+    """
+    print("Testing PcapDataset")
+    pcap_filename = os.path.join(
+        os.path.dirname(os.path.abspath(__file__)), "test_pcap", "http.pcap")
+    file_url = "file://" + pcap_filename
+    url_filenames = [file_url]
+    dataset = pcap_io.PcapDataset(url_filenames, batch=1)
+    iterator = dataset.make_initializable_iterator()
+    init_op = iterator.initializer
+    get_next = iterator.get_next()
+    with tf.compat.v1.Session() as sess:
+        sess.run(init_op)
+        v = sess.run(get_next)
+        first_packet_timestamp = v[0][0]
+        first_packet_data = v[1][0]
+        assert first_packet_timestamp == 1084443427.311224 # we know this is the correct value in the test pcap file
+        assert len(first_packet_data) == 62 # we know this is the correct packet data buffer length in the test pcap file
+
+
+        packets_total = 43 # we know this is the correct number of packets in the test pcap file
+        for packets_read in range(1, packets_total):
+            v = sess.run(get_next)
+            next_packet_timestamp = v[0][0]
+            next_packet_data = v[1][0]
+            assert next_packet_timestamp
+            assert next_packet_data
+
+
+if __name__ == "__main__":
+    test.main()
```

Comment:
Also as part of the effort to move to DatasetV2, we want the test to focus on TF 2.0 env. In TF 2.0 env, the biggest difference is that Eager execution is enabled by default. There is no need for `tf.Session` anymore.

Since we still want to support TF 1.x, we test TF 1.x in eager execution (non-default). The following will bring in eager execution in TF 1.x, and is compatible with TF 2.0 in any case:
```
if not (hasattr(tf, "version") and tf.version.VERSION.startswith("2.")):
  tf.compat.v1.enable_eager_execution()
```

With Eager execution enabled, the iteration of dataset could be simplified to
```
   for v in dataset:
      print(v)
```
And OutOfRange exception is handled automatically.

Overall, here is the test case:
```
"""
Test PcapDataset
"""
from __future__ import absolute_import
from __future__ import division
from __future__ import print_function

import os

import tensorflow as tf
if not (hasattr(tf, "version") and tf.version.VERSION.startswith("2.")):
  tf.compat.v1.enable_eager_execution()
import tensorflow_io.pcap as pcap_io # pylint: disable=wrong-import-position

def test_pcap_input():
  """test_pcap_input
  """
  print("Testing PcapDataset")
  pcap_filename = os.path.join(
      os.path.dirname(os.path.abspath(__file__)), "test_pcap", "http.pcap")
  file_url = "file://" + pcap_filename
  url_filenames = [file_url]
  dataset = pcap_io.PcapDataset(url_filenames, batch=1)

  packets_total = 0
  for v in dataset:
    (packet_timestamp, packet_data) = v
    if packets_total == 0:
      assert packet_timestamp.numpy()[0] == 1084443427.311224 # we know this is the correct value in the test pcap file
      assert len(packet_data.numpy()[0]) == 62 # we know this is the correct packet data buffer length in the test pcap file
    packets_total += 1
  assert packets_total == 43 # we know this is the correct number of packets in the test pcap file

if __name__ == "__main__":
  test.main()
```

Additional notes:
1: We use `.travis/lint.python.sh` to lint python code:
   ```
curl -o .pylint -sSL https://raw.githubusercontent.com/tensorflow/tensorflow/master/tensorflow/tools/ci_build/pylintrc

find . -name \*.py | xargs pylint --rcfile=.pylint
   ```

2. TF's eager execution does not play well with non-eager mode within the same process. For that reason, our CI will invoke twice. One for files ended with `test_xxx_eager.py` and another one for files not end with `_eager.py`.
    With pcap support DatasetV2, the file name could be changed to `tests/test_pcap_eager.py`.